definitions/scenarios/upgrade_to_katello_4_5_z.rb

```diff
@@ -0,0 +1,93 @@
+module Scenarios::Katello_4_5_z
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          feature(:katello_install) &&
+            (feature(:katello).current_minor_version == '4.5' || \
+            ForemanMaintain.upgrade_in_progress == '4.5.z')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '4.5.z'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Katello 4.5.z'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before upgrading to Katello 4.5.z'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Upgrade steps for Katello 4.5.z'
+      tags :migrations
+      run_strategy :fail_fast
+    end
+
+    def set_context_mapping
+      context.map(:assumeyes, Procedures::Installer::Upgrade => :assumeyes)
+    end
+
+    def compose
+      # The version 3.3 is of relevant Foreman version on which Katello 4.5 released
+      add_step(Procedures::Repositories::Setup.new(:version => '3.3'))
+      modules_to_enable = ["foreman:#{el_short_name}",
+                           "katello:#{el_short_name}", "pulpcore:#{el_short_name}"]
+      add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable,
+                                                       :assumeyes => true))
+      add_step(Procedures::Packages::Update.new(:assumeyes => true))
+      add_step_with_context(Procedures::Installer::Upgrade)
+      add_step(Procedures::Installer::UpgradeRakeTask)
```

Comment:
It was a bit confusing to me that this is here, but looks like there's an `if feature(:satellite)` in the UpgradeRakeTask itself.

Reply:
Removing it, considering the Foreman and Katello scenarios already do that